allauth/account/adapter.py

```diff
@@ -565,9 +565,26 @@ def is_safe_url(self, url):
 
         return url_has_allowed_host_and_scheme(url, allowed_hosts=allowed_hosts)
 
+    def request_password_reset(self, form, request: HttpRequest, **kwargs) -> str:
```

Comment:
It would be good to improve the method signature:
- `request` can be dropped -- over time the adapter methods grew inconsisently, some have `request`, some don't. Plan is to move to not requiring it, as the request is always available via `self.request`.
- An untyped/unspecific`**kwargs` is not very nice
- Question: is it wise to make this method require a `form` at all? An alternative would be for it to accept `user` instead.